spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -292,12 +296,100 @@ public void afterSingletonsInstantiated() {
 						this.applicationContext.getBean(IntegrationManagementConfigurer.MANAGEMENT_CONFIGURER_NAME,
 								IntegrationManagementConfigurer.class);
 			}
+
+			this.singletonsInstantiated = true;
 		}
 		catch (RuntimeException e) {
 			unregisterBeans();
 			throw e;
 		}
+	}
+
+	@Override
+	public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
+		if (this.singletonsInstantiated) {
```

Comment:
Since we are already at RC, perhaps we should add a `try {...} catch {...}` under here to fail soft (and log an error) instead of failing hard (just in case there are any regression issues) ?